### PR TITLE
Initiate charm modernisation

### PR DESCRIPTION
**Pull request by jat-canonical**
This commit adds basic configuration to linters and formatters. This commit also satisfies the linters and formatters. This commit also creates jobs for github actions.

### CODE REVIEW

File: .github/renovate-config.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  branchPrefix: "renovate/",
+  dryRun: null,
+  username: "renovate-release",
+  gitAuthor: "Renovate Bot <bot@renovateapp.com>",
+  onboarding: true,
+  platform: "github",
+  includeForks: true,
+  repositories: ["jat-canonical/nats-operator"],
```
Comment (morphis) [inline .github/renovate-config.js]: This is wrong and needs to become "canonical/charm-lxd-integrator"

File: .github/workflows/build-and-test.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3 # v3
+      - name: Install dependencies
+        run: python -m pip install tox
+      - name: Run tests
+        run: tox -e unit
+
+  integration-test:
+    name: Integration tests
+    runs-on: ubuntu-20.04
```
Comment (morphis) [inline .github/workflows/build-and-test.yaml:31]: Why 20.04? Lets use 22.04 if there is nothing speaking against it.

**Comment by jat-canonical** [comment]
@morphis Have a look at this one. Not sure why the GHA aren't running.

**Review by morphis — CHANGES_REQUESTED** [review]
Cam we update README and CONTRIBUTING as well as we did for the nats-operator charm?

File: .github/workflows/build-and-test.yaml (hunk 2 above)
Comment (jat-canonical) [reply .github/workflows/build-and-test.yaml:31]: the charm supports building on 20.04 to support running on 20.04 as well. Hence had to run the integration tests on 20.04.

File: README.md [hunk 3]
```diff
@@ -64,4 +71,13 @@ protocol, host and port.
...
 has been processed by the `provides` side, it should be available in the
-`registered_certificates` list on the relation.
\ No newline at end of file
+`registered_certificates` list on the relation.
+
+## Security
+Security issues in the operator can be reported through [LaunchPad](https://wiki.ubuntu.com/DebuggingSecurity#How%20to%20File) on the [Anbox Cloud](https://bugs.launchpad.net/anbox-cloud) project. Please do not file GitHub issues about security issues.
+
+## Contributing
+Please see the [Juju SDK docs](https://juju.is/docs/sdk) for guidelines on enhancements to this charm following best practice guidelines, and [CONTRIBUTING.md](https://github.com/canonical/nats-operator/blob/main/CONTRIBUTING.md) for developer guidance.
```
Comment (morphis) [inline README.md]: This is linking to the CONTRIBUTING.md of the nats-operator. We need to have our own here and link it.

**Review by morphis — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -64,4 +71,13 @@ protocol, host and port.
...
 has been processed by the `provides` side, it should be available in the
-`registered_certificates` list on the relation.
\ No newline at end of file
+`registered_certificates` list on the relation.
+
+## Security
+Security issues in the operator can be reported through [LaunchPad](https://wiki.ubuntu.com/DebuggingSecurity#How%20to%20File) on the [Anbox Cloud](https://bugs.launchpad.net/anbox-cloud) project. Please do not file GitHub issues about security issues.
+
+## Contributing
+Please see the [Juju SDK docs](https://juju.is/docs/sdk) for guidelines on enhancements to this charm following best practice guidelines, and [CONTRIBUTING.md](https://github.com/canonical/charm-lxd-integrator/blob/main/CONTRIBUTING.md) for developer guidance.
```
Comment (morphis) [inline README.md:80]: We still miss the actual CONTRIBUTING.md file in the PR

**Review by morphis — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (morphis) [reply README.md:80]: Let me also rename the branch to "main" so the link actually works.

**Review by morphis — APPROVED** [review]
Going to merge this as-is and add the `CONTRIBUTING.md` myself to see why we don't get the actions to work.

**Pull request merged**